### PR TITLE
gcp kms Keyring analysis added

### PR DESCRIPTION
**Pull request by Aryan-code-dev**

### CODE REVIEW

**Comment by mpurusottamc** [comment]
@Aryan-code-dev Please resolve merge conflicts.

File: cartography/data/jobs/cleanup/gcp_kms_cleanup.json [hunk 1]
```diff
@@ -12,6 +12,12 @@
 			"iterationsize": 100,
 			"__comment__": "Remove GCP GCPKMSCryptoKey-to-Keyrings relationships that are out of date."
 		},
+		{
+			"query": "MATCH (n:GCPBinding)<-[:ATTACHED_BINDING]-(:GCPKMSKeyRing)<-[:RESOURCE]-(:GCPLocation)<-[:RESOURCE]-(:GCPProject{id: $GCP_PROJECT_ID})<-[:OWNER]-(:CloudanixWorkspace{id: $WORKSPACE_ID}) WHERE n.lastupdated <> $UPDATE_TAG WITH n LIMIT $LIMIT_SIZE DETACH DELETE (n) return COUNT(*) as TotalCompleted",
+			"iterative": true,
+			"iterationsize": 100,
+			"__comment__": "Delete GCP KMS Keyring Bindings that no longer exist and detach them from all previously connected nodes."
```
Comment (ganeshjagdale9545) [inline cartography/data/jobs/cleanup/gcp_kms_cleanup.json:19]: create cleanup job for `ATTACHED_BINDING ` relationship

File: cartography/intel/gcp/cloudkms.py [hunk 2]
```diff
@@ -111,9 +111,9 @@ def get_kms_keyrings(kms: Resource, kms_locations: List[Dict], project_id: str)
                         key_ring['id'] = key_ring['name']
                         key_ring['key_ring_name'] = key_ring['name'].split('/')[-1]
                         key_ring['region'] = loc.get("locationId", "global")
-                        key_ring_entities, public_access = get_keyring_policy_entities(kms, key_ring, project_id)
-                        key_ring['entities'] = key_ring_entities
-                        key_ring['public_access'] = public_access
+                        #key_ring_entities, public_access = get_keyring_policy_entities(kms, key_ring, project_id)
+                        # key_ring['entities'] = key_ring_entities
+                        # key_ring['public_access'] = public_access
```
Comment (ganeshjagdale9545) [inline cartography/intel/gcp/cloudkms.py:116]: Instead of commenting code remove it

File: cartography/intel/gcp/cloudkms.py [hunk 3]
```diff
@@ -171,7 +172,53 @@ def get_keyring_policy_entities(kms: Resource, keyring: Dict, project_id: str) -
...
+    Process the GCP kms_policy_binding objects and return a flattened list of GCP bindings with all the necessary fields
+    we need to load it into Neo4j
+    :param response_objects: The return data from get_gcp_function_policy_bindings()
+    :return: A list of GCP GCP bindings
+    """
+    binding_list = []
+    for res in response_objects:
+        res['id'] = f"projects/{project_id}/function/{function_id}/role/{res['role']}"
+        binding_list.append(res)
+    return binding_list
```
Comment (ganeshjagdale9545) [inline cartography/intel/gcp/cloudkms.py:187]: in this method make changes for keryring instead of function

**Review by ganeshjagdale9545 — CHANGES_REQUESTED** [review]
few minor changes needed

File: cartography/data/jobs/cleanup/gcp_kms_cleanup.json (hunk 1 above)
Comment (Aryan-code-dev) [reply cartography/data/jobs/cleanup/gcp_kms_cleanup.json:19]: can you check i added the changes

File: cartography/intel/gcp/cloudkms.py (hunk 2 above)
Comment (Aryan-code-dev) [reply cartography/intel/gcp/cloudkms.py]: done

File: cartography/intel/gcp/cloudkms.py [hunk 4]
```diff
@@ -157,8 +157,9 @@ def get_keyring_policy_entities(kms: Resource, keyring: Dict, project_id: str) -
     try:
         iam_policy = kms.projects().locations().keyRings().getIamPolicy(resource=keyring['id']).execute()
         bindings = iam_policy.get('bindings', [])
-        entity_list, public_access = iam.transform_bindings(bindings, project_id)
-        return entity_list, public_access
+        #entity_list, public_access = iam.transform_bindings(bindings, project_id)
+        #return entity_list, public_access
```
[the remark this answers is not in the document]
Comment (Aryan-code-dev) [reply cartography/intel/gcp/cloudkms.py]: done

File: cartography/intel/gcp/cloudkms.py (hunk 3 above)
Comment (Aryan-code-dev) [reply cartography/intel/gcp/cloudkms.py:182]: done...sorry missed in hurry

**Review by ganeshjagdale9545 — APPROVED** [review]
Looks good

**Pull request merged**